platforms/gke/base/core/deploy.sh

```diff
@@ -17,19 +17,20 @@ set -o errexit
 
 start_timestamp=$(date +%s)
 
-terraservices=("networking" "container_cluster" "container_node_pool" "gke_enterprise/fleet_membership" "workloads/kueue")
 # Disable gke_enterprise/servicemesh due to b/376312292
-#terraservices=("networking" "container_cluster" "container_node_pool" "gke_enterprise/fleet_membership" "gke_enterprise/servicemesh" "workloads/kueue")
+declare -a terraservices=${TERRASERVICES:-("networking" "container_cluster" "container_node_pool" "gke_enterprise/fleet_membership" "workloads/kueue")}
 
 source ${ACP_PLATFORM_BASE_DIR}/_shared_config/scripts/set_environment_variables.sh ${ACP_PLATFORM_BASE_DIR}/_shared_config
 
 cd ${ACP_PLATFORM_CORE_DIR}/initialize &&
     echo "Current directory: $(pwd)" &&
+    sed -i "s/^\([[:blank:]]*bucket[[:blank:]]*=\).*$/\1 \"${ACP_TERRAFORM_BUCKET_NAME}\"/" ${ACP_PLATFORM_CORE_DIR}/initialize/backend.tf.bucket &&
     export STATE_MIGRATED="false" &&
     if gcloud storage ls gs://${ACP_TERRAFORM_BUCKET_NAME}/terraform/initialize/default.tfstate &>/dev/null; then
-        sed -i "s/^\([[:blank:]]*bucket[[:blank:]]*=\).*$/\1 \"${ACP_TERRAFORM_BUCKET_NAME}\"/" ${ACP_PLATFORM_CORE_DIR}/initialize/backend.tf.bucket &&
-            cp backend.tf.bucket backend.tf &&
-            export STATE_MIGRATED="true"
+        if [ ! -f ${ACP_PLATFORM_CORE_DIR}/initialize/backend.tf ]; then
+            cp backend.tf.bucket backend.tf
+        fi
+        export STATE_MIGRATED="true"
     fi
```

Comment:
Can we handle this logic using Terraform as well?

Reply:
We can't because you need the bucket to be set before you run Terraform. I think the other option would be to set it as a command line argument but it accomplishes the same thing. This is really only for testing or cases where you don't commit the `backend.tf` file to the repository.